### PR TITLE
Update Grafana datasource type

### PR DESCRIPTION
**Pull request by Gary-H9**
After working on issue [335](https://github.com/grafana/terraform-provider-grafana/issues/335) I mistakenly added the datasource [type ](https://github.com/grafana/terraform-provider-grafana/pull/340/files#diff-489a9f185703571407b8ec93e9c9fb4043ab01e6fd3b081728b5032facd852b2R324) as `'github'` and not `'grafana-github-datasource'` in the [PR](https://github.com/grafana/terraform-provider-grafana/pull/340). 

This PR seeks to rectify this (github datasources are not created properly as it currently stands) and also rectifies a couple of typos I noted when working on this. 🙏

I will further improve the docs in the near future to include a github example.

### CODE REVIEW

**Comment by Gary-H9** [comment]
I've made further changes, some variable names were incorrect:
* `access_token ` -> `accessToken`
* `github_url` -> `githubUrl`
* `secure_json_data `-> `secureJsonData`

I've also added a github datasource example to the documentation.

**Comment by Gary-H9** [comment]
@julienduchesne - could you help me out here? I'm not overly familiar with Go and I'm not sure why I have numerous failing checks.

**Comment by julienduchesne** [comment]
You are right about the type being `grafana-github-datasource` but the attributes are bound correctly already:
- `githubUrl`: https://github.com/grafana/grafana-api-golang-client/blob/master/datasource.go#L99
- `accessToken`: https://github.com/grafana/grafana-api-golang-client/blob/master/datasource.go#L256

What you are seeing is just a syntax translation. Terraform takes in snake case but the API takes in camel case. The translation is done in the code

I think just changing `github` for `grafana-github-datasource` in the examples should do the trick

File: examples/resources/grafana_data_source/resource.tf [hunk 1]
```diff
@@ -50,3 +50,16 @@ resource "grafana_data_source" "stackdriver" {
     private_key = "-----BEGIN PRIVATE KEY-----\nprivate-key\n-----END PRIVATE KEY-----\n"
   }
 }
+
+resource "grafana_data_source" "github" {
```
Comment (julienduchesne) [inline examples/resources/grafana_data_source/resource.tf]: Since this requires an external plugin, I wouldn't add it as an example here

Comment (Gary-H9) [reply examples/resources/grafana_data_source/resource.tf]: Understood thanks, I'll remove this.

File: grafana/resource_data_source_test.go [hunk 2]
```diff
@@ -448,21 +448,21 @@ func TestAccDataSource_basic(t *testing.T) {
...
 			resource "grafana_data_source" "github" {
-				type = "github"
+				type = "grafana-github-datasource"
 				name = "github"
 				json_data {
 					github_url = "https://github.com"
 				}
 				secure_json_data {
-					access_token = "token for github"
+					accessToken = "token for github"
```
Comment (julienduchesne) [inline grafana/resource_data_source_test.go]: Here also, `access_token` refers to the Terraform format. So it was already correct

Comment (Gary-H9) [reply grafana/resource_data_source_test.go]: Understood thanks again. Latest push corrects this.

**Comment by Gary-H9** [comment]
Thanks for your help @julienduchesne. I've removed the erroneous alterations I made, I think I'd just been looking at this too long and had gotten confused. `github_url`, `secure_json_data` and `access_token` are now in the correct format.

I've also removed the addition to the documentation as per your note.

**Comment by julienduchesne** [comment]
You were right that it wasn't working @Gary-H9. The issue was that the values weren't being passed to the API model. See https://github.com/grafana/terraform-provider-grafana/pull/422/files#diff-96f3c606e2c6e71c26a6e75f391fc8f850f8200403cc2d652b663f43cbfc8386R665

I fixed it and added another test. Thanks!

**Review by julienduchesne — APPROVED** [review]

**Comment by Gary-H9** [comment]
Thanks @julienduchesne

**Pull request merged**